### PR TITLE
DEV-366

### PR DESCRIPTION
**Pull request by teye**
refactor team service

### CODE REVIEW

File: service-registration/src/main/java/sg/ncl/service/registration/AppConfig.java.orig [hunk 1]
```diff
@@ -0,0 +1,19 @@
+package sg.ncl.service.registration;
```
Comment (ChrisZhong) [inline service-registration/src/main/java/sg/ncl/service/registration/AppConfig.java.orig]: Please do not check in `*.orig` files

File: service-team/src/main/java/sg/ncl/service/team/data/jpa/TeamMemberEntity.java [hunk 2]
```diff
@@ -17,7 +17,7 @@
     @Id
     @GeneratedValue
     @Column(name = "id", nullable = false, unique = true, updatable = false)
-    private Long id;
+    private String id;
```
Comment (ChrisZhong) [inline service-team/src/main/java/sg/ncl/service/team/data/jpa/TeamMemberEntity.java]: Why was this changed to a `String`?

Comment (teye) [inline service-team/src/main/java/sg/ncl/service/team/data/jpa/TeamMemberEntity.java]: The rest of the id's are implemented in String.

Comment (ChrisZhong) [inline service-team/src/main/java/sg/ncl/service/team/data/jpa/TeamMemberEntity.java]: Please change it back to `Long`; there are only 2 IDs that are `String`: user and team. All other IDs should be kept to the default of `Long`. The reason the user and team IDs are `String` is because we expose the IDs through APIs like `GET /users/{id}`. A randomly generated `String` is harder to guess than the default sequential `Long`.

File: service-team/src/main/java/sg/ncl/service/team/logic/TeamService.java [hunk 3]
```diff
@@ -41,33 +41,33 @@ public TeamEntity save(TeamEntity team) {
...
+        final List<Team> result = new ArrayList<>();
+        for (TeamEntity teamEntity : teamRepository.findAll()) {
+            result.add(teamEntity);
         }
         return result;
     }
 
-    public List<TeamEntity> getPublic() {
+    @Transactional
+    public List<Team> getPublic() {
```
Comment (ChrisZhong) [inline service-team/src/main/java/sg/ncl/service/team/logic/TeamService.java]: Refactor method to `getPublicTeams()`

File: service-team/src/main/java/sg/ncl/service/team/logic/TeamService.java [hunk 4]
```diff
@@ -41,33 +41,33 @@ public TeamEntity save(TeamEntity team) {
...
-            throw new TeamNotFoundException();
+        for (TeamEntity teamEntity : result) {
+            teamList.add(teamEntity);
         }
+        return teamList;
+    }
 
-        return one;
+    @Transactional
+    public Team findTeam(final String id) {
```
Comment (ChrisZhong) [inline service-team/src/main/java/sg/ncl/service/team/logic/TeamService.java]: Refactor method to `getById`

File: service-team/src/main/java/sg/ncl/service/team/logic/TeamService.java [hunk 5]
```diff
@@ -41,33 +41,33 @@ public TeamEntity save(TeamEntity team) {
...
+        return findTeamEntity(id);
     }
 
-    public TeamEntity getName(String name) {
+    @Transactional
+    public Team getName(String name) {
```
Comment (ChrisZhong) [inline service-team/src/main/java/sg/ncl/service/team/logic/TeamService.java]: Refactor method to `getByName()`

File: service-team/src/main/java/sg/ncl/service/team/web/TeamsController.java [hunk 6]
```diff
@@ -34,50 +27,45 @@ protected TeamsController(final TeamService teamService) {
 
     @RequestMapping(path = "", method = RequestMethod.POST)
     @ResponseStatus(value = HttpStatus.CREATED)
-    public void addTeam(@RequestBody @Valid TeamEntity team) {
-        teamService.save(team);
+    public void createTeam(@RequestBody @Valid TeamInfo team) {
+        teamService.createTeam(team);
```
Comment (ChrisZhong) [inline service-team/src/main/java/sg/ncl/service/team/web/TeamsController.java]: This should return the `Team` that was created

File: service-team/src/main/java/sg/ncl/service/team/web/TeamsController.java [hunk 7]
```diff
@@ -34,50 +27,45 @@ protected TeamsController(final TeamService teamService) {
...
 
     @RequestMapping(path = "/public", method = RequestMethod.GET)
     @ResponseStatus(code = HttpStatus.OK)
-    public List<TeamInfo> getPublicTeams() {
-        List<TeamEntity> teamEntityList = teamService.getPublic();
-        List<TeamInfo> teamInfoList = new ArrayList<>();
-
-        for (TeamEntity teamEntity : teamEntityList) {
-            teamInfoList.add(new TeamInfo(teamEntity));
+    public List<Team> getPublicTeams() {
```
Comment (ChrisZhong) [inline service-team/src/main/java/sg/ncl/service/team/web/TeamsController.java]: This needs to be generalized to `getByVisibility(@RequestParam("visiblity") TeamVisibility visibility)`
Use query parameters (e.g., `GET /teams?visibility=PUBLIC`)

File: service-team/src/main/java/sg/ncl/service/team/web/TeamsController.java [hunk 8]
```diff
@@ -34,50 +27,45 @@ protected TeamsController(final TeamService teamService) {
...
+            result.add(new TeamInfo(team));
         }
-        return teamInfoList;
+        return result;
     }
 
     @RequestMapping(path = "/name/{name}", method = RequestMethod.GET)
     @ResponseStatus(code = HttpStatus.OK)
-    public TeamInfo getByName(@PathVariable String name) {
+    public Team getByName(@PathVariable String name) {
```
Comment (ChrisZhong) [inline service-team/src/main/java/sg/ncl/service/team/web/TeamsController.java]: Needs to change from `@PathVariable` to `@RequestParam` so that the format is `\teams?name=NAME`

File: service-team/src/main/java/sg/ncl/service/team/web/TeamsController.java [hunk 9]
```diff
@@ -34,50 +27,45 @@ protected TeamsController(final TeamService teamService) {
...
 
     @RequestMapping(path = "/{id}", method = RequestMethod.PUT)
     @ResponseStatus(code = HttpStatus.ACCEPTED)
-    public void updateTeam(@PathVariable String id, @RequestBody TeamEntity teamEntity) {
-        teamService.update(teamEntity);
+    public void updateTeam(@PathVariable String id, @RequestBody TeamInfo team) {
+        teamService.updateTeam(id, team);
     }
 
     @RequestMapping(path = "/addUserToTeam/{id}", method = RequestMethod.POST)
```
Comment (ChrisZhong) [inline service-team/src/main/java/sg/ncl/service/team/web/TeamsController.java]: Path should be `/{id}/members`

**Comment by ChrisZhong** [comment]
Needs more work
